src/d/actor/d_a_shand.cpp

```diff
@@ -4,91 +4,488 @@
+            actor->field_31C[19].mPos = actor->field_2D4 = actor->field_2C8 = local94.GetCross();
+        actor->field_318--;
+    }
+
+    switch(actor->field_2BA){
+        case 0:
+            if(std::abs(actor->field_31C[19].mPos.y - actor->field_2D4.y) < 10.0f){
+                actor->field_2BA = 1;
+                *actor->field_314 = 2;
+            }
+        case 1:
+            actor->field_2D4 = actor->field_2C8;
+    }
+
+    float y_diff = std::abs(actor->home.pos.y - actor->field_2C8.y);
+    cLib_addCalc2(&actor->field_2F4,   y_diff * (REG14_F(11) + 0.05f), 0.1f, 1.0f);
+    cLib_addCalc2(&actor->field_300, REG14_F(12) + 10.0f, 0.1f, 0.5f);
+    control1(actor);
+    control2(actor);
+    control3(actor);
 }
 
 /* 00001428-00001508       .text cut__FP11shand_class */
-void cut(shand_class*) {
-    /* Nonmatching */
+void cut(shand_class* actor) {
+    cLib_addCalc2(&actor->field_2F4, REG8_F(12) + 20.0f, 0.1f, 0.5f);
+    cut_control(actor);
+    cut_control3(actor);
+    if(actor->field_2BC[0] == 0 && actor->field_30C->health != 0){
+        actor->mState = 0;
+        actor->field_2BA = 0;
+        actor->field_2F0 = 0.0f;
+        actor->field_300 = 0.0f;
+        fopAcM_seStart(actor, JA_SE_OBJ_VINE_S_RECOVER, 0);
+    }
 }
 
 /* 00001508-00001DAC       .text hand_move__FP11shand_class */
-void hand_move(shand_class*) {
-    /* Nonmatching */
+void hand_move(shand_class* actor) {
+    shand_s* shand_i = actor->field_31C;
+    actor->field_30C = static_cast<shand_class*>(fopAcM_SearchByID(actor->field_308));
+    if(actor->field_30C != NULL){
+        actor->current.pos = *actor->field_310;
+        actor->current.angle.y = actor->home.angle.y + actor->field_30C->shape_angle.y + REG14_S(3);
+        switch(actor->mState){
+            case 0:
+                normal(actor);
+                actor->attention_info.flags = fopAc_Attn_LOCKON_MISC_e;
+                if(actor->field_30C->health == 0){
+                    actor->mState = 1;
+                    actor->field_2BA = 0;
+                }
+                break;
+
+            case 1: 
+                if((fopAcM_GetParam(actor) & 0xFF) != 1){
+                    dBgS_GndChk local_ac;
+                    float chk_pos_x = actor->current.pos.x, chk_pos_y = actor->current.pos.y, chk_pos_z = actor->current.pos.z;
+                    chk_pos_y -= 100.0f;
+                    local_ac.GetPointP()->set(chk_pos_x, chk_pos_y, chk_pos_z);
+                    actor->field_CE0 = dComIfG_Bgsp()->GroundCross(&local_ac);
+                    
+                    dBgS_ObjGndChk_Spl local_100;
+                    chk_pos_x = actor->current.pos.x, chk_pos_y = actor->current.pos.y, chk_pos_z = actor->current.pos.z;
+                    chk_pos_y += 200.0f;
+                    local_100.GetPointP()->set(chk_pos_x, chk_pos_y, chk_pos_z);
+                    float gnd_cross = dComIfG_Bgsp()->GroundCross(&local_100) + 10.0f;
+                    if(gnd_cross != -1e09f){
```

Comment:
There's a define for this
```suggestion
                    if(gnd_cross != C_BG_MIN_HEIGHT){
```